cmd/tempo-vulture/main.go

```diff
@@ -226,43 +104,66 @@ func generateRandomInt(min int64, max int64) int64 {
 	return number
 }
 
-func queryTempoAndAnalyze(baseURL string, seed int64, traceID string) (traceMetrics, error) {
+func queryTempoAndAnalyze(baseURL string, complete completeTrace) (traceMetrics, error) {
 	tm := traceMetrics{
 		requested: 1,
 	}
 
-	logger := logger.With(
-		zap.String("query_trace_id", traceID),
-		zap.String("tempo_query_url", baseURL+"/api/traces/"+traceID),
-		zap.Int64("seed", seed),
+	traceID := complete.traceID
+
+	log := logger.With(
+		zap.String("query_trace_id", traceID.String()),
+		zap.String("tempo_query_url", baseURL+"/api/traces/"+traceID.String()),
 	)
-	logger.Info("querying Tempo")
+	log.Info("querying Tempo")
 
-	trace, err := util.QueryTrace(baseURL, traceID, tempoOrgID)
+	trace, err := util.QueryTrace(baseURL, traceID.String(), tempoOrgID)
 	if err != nil {
 		if err == util.ErrTraceNotFound {
 			tm.notFound++
 		} else {
 			tm.requestFailed++
 		}
-		logger.Error("error querying Tempo", zap.Error(err))
+		log.Error("error querying Tempo", zap.Error(err))
 		return tm, err
 	}
 
 	if len(trace.Batches) == 0 {
-		logger.Error("trace contains 0 batches")
+		log.Error("trace contains 0 batches")
 		tm.notFound++
 	}
 
 	// iterate through
 	if hasMissingSpans(trace) {
-		logger.Error("trace has missing spans")
+		log.Error("trace has missing spans")
 		tm.missingSpans++
 	}
 
+	count := spanCount(trace)
+	if count != complete.spanCount {
```

Comment:
excellent improvement and we will merge as is, but the long term vision is to regenerate the entire trace from the seed and check all events/attributes/spans.  i.e. revalidate the entire trace.

Reply:
One thing I couldn't figure out was why attributes weren't being sent.  There is this notion of "recording" or not, and the otlp code seems to skip doing anything with attributes if the exporter isn't recording.  If we can figure out how to enable that bit, then checking the attributes should be extensible from the structure that is in place now.

Reply:
There is another way to get the attributes on the trace.  I'll see if I can get something in place to include more detail that we can check for in the trace.

Reply:
I've updated this to inject some randomness into the attributes and ensure they get validated on the query.